File: TypeScript/ProjectEuler6.ts
class List {
    private list: number[] = [];

    constructor(to:number) {
        for (var i = 1; i < to + 1; i++) {
                this.list.push(i);
        }
    }

    getList() {
        return this.list;
    }
}


var list = new List(20).getList();

var sumSquare = (data:number[]) => {
    return data.reduce((x, y) => x + (y * y));
}

var squareSum = (data:number[]) => {
    return Math.pow(data.reduce((x, y) => x + y),2);
}

console.log(squareSum(list) - sumSquare(list));
while (true) {
    
}

File: Scala/ProjectEuler6.scala
val list = List.range(1,21)

val sumSquare = (data:List[Int]) =>{
  data.fold(1)((acc,x) =>  acc + (x * x))
}

val squareSum = (data:List[Int]) => {
  Math.pow(data.fold(0)(_+_),2)
}

val sum = sumSquare(list)
val sum2 = squareSum(list)

val result = sum2 - sum
